casadm/cas_lib.c

```diff
@@ -547,6 +550,42 @@ void print_slow_atomic_cache_start_info(const char *device_path)
 	}
 }
 
+/**
+  * Save to dest an absolute device file path of src.
+  * Return number of characters copied to dest if succeed, negative value if failed.
+  */
+static int get_abs_path(char* dest, size_t dest_len, const char* src, size_t src_len)
+{
+	int path_len = -FAILURE;
+	char *dir_name, *dev_name;
+	char *dev = strndup(src, src_len);	// strdup creates hidden malloc
+	if (!dev)			// basename/dirname may modify the source and
+		goto dev_err;		// segfault when called with a static string
+
+	char *dir = strndup(src, src_len);
+	if (!dir)
+		goto dir_err;
+
+	dir_name = realpath(dirname(dir), NULL); // realpath creates hidden malloc
+	if (!dir_name)
+		goto dir_name_err;
+
+	dev_name = basename(dev);
+	if (!dev_name)
+		goto dir_name_err;
+
+	path_len = snprintf(dest, dest_len, "%s/%s", dir_name, dev_name);
+
+dir_name_err:
+	free(dir_name);
+dir_err:
+	free(dir);
+dev_err:
+	free(dev);
```

Comment:
In case of error in :561 , the `dev` pointer will be set to NULL, so you don't have call `free()`. Lines 583 and 584 should be in reverse order. The same for 581 & 582  and 579 & 580

Reply:
Thanks, I wasn't sure about it..